cupy/testing/array.py

```diff
@@ -128,6 +129,70 @@ def assert_array_list_equal(xlist, ylist, err_msg='', verbose=True):
             verbose=verbose)
 
 
+def _get_underlying_bytes(a):
+    if isinstance(a, cupy.ndarray):
+        # copy the array to host memory while preserving order
+        buf = ctypes.create_string_buffer(a.nbytes)
+        a.data.copy_to_host(ctypes.c_void_p(ctypes.addressof(buf)), a.nbytes)
+        arr = bytearray(a.nbytes)
+        arr[:] = buf.raw
+        return arr
+    else:
+        return ctypes.string_at(a.ctypes.data, a.nbytes)
+
+
+def assert_array_exactly_equal(x, y, err_msg='', verbose=True):
```

Comment:
Does this function compare garbage bytes between elements?
Is there any point to do that? How about comparing strides instead?